address.md

```diff
@@ -0,0 +1,262 @@
+# Address
+
+A Filecoin address is an identifier that refers to an actor in the Filecoin state. All actors (miner actors, the storage market actor, account actors) have an address. This address encodes information about the network to which an actor belongs, the specific type of address encoding, the address payload itself, and a checksum. The goal of this format is to provide a robust address format that is both easy to use and resistant to errors.
+
+#### Design criteria
+
+1. **Identifiable**: The address must be easily identifiable as a Filecoin address.
+2. **Reliable**: Addresses must provide a mechanism for error detection when they might be transmitted outside the network.
+3. **Upgradable**: Addresses must be versioned to permit the introduction of new address formats.
+4. **Compact**: Given the above constraints, addresses must be as short as possible.
+
+## Specification
+
+A Filecoin address is a byte string consisting of:
+* A **network indicator** that identifies which network this address belongs to.
+* A **protocol indicator** that identifies the type and version of this address.
+* The **payload** used to uniquely identify the actor according to the protocol.
+* A **checksum** verifies the address is valid.
+
+```
+|---------|----------|---------|----------|
+| network | protocol | payload | checksum |
+|---------|----------|---------|----------|
+| 1 byte  |  1 byte  | n bytes |  4 bytes |
+```
+
+An example of a Filecoin address in golang:
+
+```go
+type Address struct {
+    // 0: main-net
+    // 1: test-net
+    network byte
+    
+    // 0: ID
+    // 1: Blake2b160-Hash of secp256k1 Public Key
+    // 2: Blake2b160-Hash of Actor creation data
+    // 3: BLS Public Key
+    protocol byte
+    
+    // raw bytes containing the data associated with protocol
+    payload []byte
+}
+```
+
+#### Network Indicator
+
+The **network** byte identifies the network the address belongs to:
+
+- `0` : Mainnet
+- `1` : Testnet
```

Comment:
We don't need byte identifiers for these, these are just relevant to encoded addresses